optaplanner-examples/src/test/java/org/optaplanner/examples/conferencescheduling/solver/ConferenceSchedulingScoreRulesXlsxTest.java

```diff
@@ -0,0 +1,238 @@
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.poi.xssf.usermodel.XSSFCell;
+import org.apache.poi.xssf.usermodel.XSSFSheet;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.junit.Before;
+import org.junit.Test;
+import org.optaplanner.core.api.solver.SolverFactory;
+import org.optaplanner.core.impl.domain.solution.cloner.FieldAccessingSolutionCloner;
+import org.optaplanner.core.impl.domain.solution.descriptor.SolutionDescriptor;
+import org.optaplanner.examples.common.persistence.AbstractXlsxSolutionFileIO;
+import org.optaplanner.examples.conferencescheduling.app.ConferenceSchedulingApp;
+import org.optaplanner.examples.conferencescheduling.domain.ConferenceSolution;
+import org.optaplanner.examples.conferencescheduling.domain.Room;
+import org.optaplanner.examples.conferencescheduling.domain.Talk;
+import org.optaplanner.examples.conferencescheduling.domain.Timeslot;
+import org.optaplanner.examples.conferencescheduling.persistence.ConferenceSchedulingXlsxFileIO;
+import org.optaplanner.test.impl.score.buildin.hardsoft.HardSoftScoreVerifier;
+
+public class ConferenceSchedulingScoreRulesXlsxTest {
+
+    private ConferenceSolution currentSolution;
+    private String currentPackage;
+    private String currentConstraint;
+    private String currentSheetName;
+    private int expectedHardScore, expectedSoftScore;
+
+    private final String testFileName = "testConferenceSchedulingScoreRules.xlsx";
+    private HardSoftScoreVerifier<ConferenceSolution> scoreVerifier = new HardSoftScoreVerifier<>(
+            SolverFactory.createFromXmlResource(ConferenceSchedulingApp.SOLVER_CONFIG));
+    private testConferenceSchedulingScoreRulesReader testFileReader;
+
+    @Before
+    public void setup() {
+        File testFile = new File(getClass().getResource(testFileName).getFile());
+        try (InputStream in = new BufferedInputStream(new FileInputStream(testFile))) {
+            XSSFWorkbook workbook = new XSSFWorkbook(in);
+            this.currentSolution = new ConferenceSchedulingXlsxFileIO().read(testFile);
+            this.testFileReader = new testConferenceSchedulingScoreRulesReader(workbook);
+        } catch (IOException | RuntimeException e) {
+            throw new IllegalStateException("Failed reading inputSolutionFile ("
+                                                    + testFileName + ").", e);
+        }
+    }
+
+    @Test
+    public void testRules() {
+        while ((currentSolution = testFileReader.nextSolution()) != null) {
+            //TODO: give a clue which test sheet is failing
+            System.out.println(currentSheetName);
```

Comment:
Test and production code cannot contain System.out's. It's ok to use them locally, just remove them before pushing.
If they are useful for others, use a logger instead, so it can be diverted, slienced etc through logback.xml

Motivation: a "mvn clean install" runs **all** tests. problems etc should stick out like a soar thumb. The higher the test verbosity, the harder it is too see the real interesting output. If we have 1000 tests and all output 5 lines, then 5k lines scroll by and we only read the last 50 lines or so. Stacktraces dissappear in the output.

Reply:
@ge0ffrey I agree. That said, a permanent solution to that problem, which doesn't rely on humans collaborating, is to configure Surefire to redirect the stdout to a file. It can do that per-test, and it's very handy, as you can archive that with Jenkins.